lib/myxql/connection.ex

```diff
@@ -529,14 +529,23 @@ defmodule MyXQL.Connection do
   defp queries_get(%{queries: nil}, _), do: nil
   defp queries_get(_state, %Query{name: ""}), do: nil
 
-  defp queries_get(state, %Query{cache: :reference, name: name} = query) do
+  defp queries_get(state, %Query{cache: :reference, name: name}) do
```

Comment:
@josevalim This only needs to check `statement_id` but it becomes a bit awkward because the `cache: :statement` version needs to lookup the `{statement, num_params, statement_id, ref}` tuple.

So to optimize this I think you'd have to store `{name, statement_id, statement_id, ref}` for reference queries and `{name, statement_id, {statement, num_params, statement_id, ref}, ref` for statement queries. Then reference queries lookup the 2nd element and statement queries lookup the 3rd.

I think the memory usage of integers/references is small enough that the duplicated info is worth not having to copy the statement. But I'm definitely not an expert. Please let me know your opinion.

For reference, Postgrex duplicates the reference to avoid copying the statement. But they don't have the `statement_id` to worry about.

Reply:
If we need to lookup multiple keys, perhaps we can use `:ets.lookup` instead? To avoid the duplication?

Reply:
updated. thank you, this looks nicer now.